enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx

```diff
@@ -78,9 +78,12 @@ export const InteractiveQuestionProvider = ({
       const saveContext = { isNewQuestion: true };
 
       await onBeforeSave?.(question, saveContext);
-      await handleCreateQuestion(question);
-      onSave?.(question, saveContext);
-      await loadQuestion();
+
+      const createdQuestion = await handleCreateQuestion(question);
+      onSave?.(createdQuestion, saveContext);
+
+      // Set the latest saved question object to update the question title.
+      replaceQuestion(createdQuestion);
```

Comment:
We used to call `loadQuestion` here which actually just reloads the original question, which doesn't match the expectation from the main Metabase app. This instead gets the created question from the save handler, passes it to `onSave` and replaces the current question / originalQuestion with the newly created question.

This allows the save modal to show the correct question's title, as well as adding the "- Modified" suffix when re-saving the question to a new question. Refer to the e2e test for the expected title change behavior.